src/main/java/analysis/GoblintAnalysis.java

```diff
@@ -192,4 +197,35 @@ public ProcessResult runCommand(File dirPath, String[] command) throws IOExcepti
     }
 
 
+    /**
+     * Method for creating an observer for Goblint configuration file.
+     * So that the server could be restarted when the configuration file is changed.
+     *
+     * @return The FileAlterationObserver of project root directory.
+     */
+
+    public FileAlterationObserver createGoblintConfObserver() {
+        FileAlterationObserver observer = new FileAlterationObserver(System.getProperty("user.dir"));
+        
+        observer.addListener(new FileAlterationListenerAdaptor() {        
+            @Override
+            public void onFileChange(File file) {
+                if (file.getName().equals(goblintServer.getGoblintConf())) {
```

Comment:
Since this is just watching a single file, not a whole directory (tree), there should probably be a `FileFilter` to just limit the watching to that.